src/community/jdbcstore/src/main/java/org/geoserver/jdbcstore/JDBCResourceStore.java

```diff
@@ -0,0 +1,375 @@
+            assert(entry != null);
+            this.entry = entry;
+        }
+        
+        @Override
+        public String path() {
+            return entry.toString();
+        }
+
+        @Override
+        public String name() {
+            return entry.getName();
+        }
+        
+        private InputStream getIStream() {
+            return entry.getContent();            
+        }
+                
+        @Override
+        public InputStream in() {
+            final Lock lock = lock();
+            try {
+                entry.createResource();
+                return getIStream();
+            } finally {
+                lock.release();
+            }
+        }
+
+        @Override
+        public OutputStream out() {
+            List<ResourceNotification.Event> events = SimpleResourceWatcher.createEvents(this,
+                    ResourceNotification.Kind.ENTRY_CREATE);
+            if (entry.createResource()) {
+                resourceWatcher.changed(new ResourceNotification(path(), ResourceNotification.Kind.ENTRY_CREATE, 
+                        System.currentTimeMillis(), events));
+            }
+            try {
+                return new CachingOutputStreamWrapper(File.createTempFile("out.", entry.getName()));
+            } catch (IOException ex) {
+                throw new IllegalStateException(ex); 
+            }
+        }
+
+        @Override
+        public File file() {
+            if (getType() == Type.DIRECTORY) {
+                throw new IllegalStateException("Directory (not a file)");
+            }
+            final Lock lock = lock();
+            try {
+                return cache.cache(this, false);
+            } catch(IOException ex) {
+                throw new IllegalStateException(ex);
+            } finally {
+                lock.release();
+            }
+        }
+
+        @Override
+        public File dir() {
+            if (getType() == Type.RESOURCE) {
+                throw new IllegalStateException("File (not a directory)");
+            }
+            final Lock lock = lock();
+            try {
+                return cache.cache(this, true);
+            } catch(IOException ex) {
+                throw new IllegalStateException(ex);
+            } finally {
+                lock.release();
+            }
+        }
+
+        @Override
+        public long lastmodified() {
+            Timestamp ts = entry.getLastModified();
+            return ts == null ? 0L : ts.getTime();
+        }
+
+        @Override
+        public Resource parent() {
+            return new JDBCResource(entry.getParent());
+        }
+
+        @Override
+        public Resource get(String resourcePath) {
+            return JDBCResourceStore.this.get(path() + "/" + resourcePath);
+        }
+
+        @Override
+        public List<Resource> list() {
+            if (getType() != Type.DIRECTORY) {
+                return Collections.EMPTY_LIST;
```

Comment:
If you add ```@SuppressWarnings("unchecked")``` to this method, this entire module will have no warnings at all.  :-)